src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -851,8 +849,15 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
 
         case tpnme.RuntimeAnnotationATTR =>
           val numAnnots = u2
+          val annots = new ListBuffer[AnnotationInfo]
           for (n <- 0 until numAnnots; annot <- parseAnnotation(u2))
-            sym.addAnnotation(annot)
+            annots += annot
+          /* `sym.withAnnotations(annots)`, like `sym.addAnnotation(annot)`, prepends,
+           * so if we parsed in classfile order we would wind up with the annotations
+           * in reverse order in `sym.annotations`. Instead we just read them out the
+           * other way around, for now. TODO: sym.addAnnotation add to the end?
+           */
+          sym.setAnnotations(sym.annotations ::: annots.toList)
```

Comment:
How is `sym.annotations` non-empty here?

Reply:
It might be.  Why is that a problem?  Maybe I don't understand what you mean here.

Reply:
It's not a problem; I'm just wondering why `sym setAnnotations annots.toList` is insufficient. Doesn't matter though.